### PR TITLE
Availability product variant

### PR DESCRIPTION
**Pull request by mamazu**
Replaces #648

### CODE REVIEW

**Review by lchrusciel — APPROVED** [review]
Seems ok, but probably validation to expose nice 400 error to the customer should be added as well

**Comment by lchrusciel** [comment]
Thanks, @mamazu! :tada:

**Comment by mamazu** [comment]
The idea behind this was that it was optional to add this feature. I wasn't sure it it would be a BC break to do so.

File: src/Modifier/OrderModifier.php [hunk 1]
```diff
@@ -70,4 +85,15 @@ private function getCartItemToModify(OrderInterface $cart, ProductVariantInterfa
 
         return null;
     }
+
+    private function checkCartQuantity(ProductVariantInterface $productVariant, int $targetQuantity): void
+    {
+        if($this->availabilityChecker === null) {
```
Comment (diimpp) [inline src/Modifier/OrderModifier.php:91]: Missing whitespace between `if` and `(`. ecs should have caught that, strange.

**Comment by diimpp** [comment]
Missing DI definition with actual service injection.
I think this is not a BC, it's a part of standard sylius behavior and only expected to function this way.

We should check for enabled/disabled over here as well.

Maybe we can introduce a point of extension over here, ShopApi AvailabiltyCheckerInterface with CompositeAvailabilityChecker, which will check for Inventory availability, enabled/disabled and common use case for my projects -- archived products/variants

Sounds good? I already have such system in place for my project, I can do follow PR.

**Pull request merged**